docs/ref/models/querysets.txt

```diff
@@ -2474,6 +2474,51 @@ each model instance (if the database normally supports it).
     In older versions, enabling the ``update_conflicts`` parameter prevented
     setting the primary key on each model instance.
 
+.. versionadded:: 5.1
+
+    The ``where_clause`` parameter was added.
+
+    The ``where_clause`` parameter allows specifying a conditional SQL ``WHERE`` 
+    clause for the ``DO UPDATE SET`` part of the bulk insert query. This feature
+    is useful for performing conditional updates during conflict resolution in bulk 
+    insert operations. It is currently supported only for the SQLite and PostgreSQL backends.
+
+    The syntax of this clause should be valid SQL for the database backend being used.
+    For instance, in SQLite, you might use SQLite functions in your ``WHERE`` clause,
+    whereasin PostgreSQL, you can utilize PostgreSQL-specific syntax and functions.
+
+    Example for SQLite:
+
+.. code-block:: pycon
+
+    >>> items = [
+    ...     Item(name="Item 1", last_updated=timezone.now() - timedelta(days=1)),
+    ...     Item(name="Item 2", last_updated=timezone.now()),
+    ... ]
+    >>> Item.objects.bulk_create(
+    ...     items,
+    ...     update_conflicts=True,
+    ...     update_fields=["name", "last_updated"],
+    ...     unique_fields=["id"],
+    ...     where_clause="datetime('now') > datetime(last_updated)",
+    ... )
+
+    Example for PostgreSQL:
+
+.. code-block:: pycon
+
+    >>> items = [
+    ...     Item(name="Item 1", last_updated=timezone.now() - timedelta(days=1)),
+    ...     Item(name="Item 2", last_updated=timezone.now()),
+    ... ]
+    >>> Item.objects.bulk_create(
+    ...     items,
+    ...     update_conflicts=True,
+    ...     update_fields=["name", "last_updated"],
+    ...     unique_fields=["id"],
+    ...     where_clause="EXCLUDED.last_updated > last_updated",
+    ... )
+
```

Comment:
Again, for this change to be accepted, we need to support providing expressions to `where_clause` rather than having people write backend-specific SQL with risk of SQL injection.